sys/riscv/riscv/trap.c

```diff
@@ -470,6 +470,13 @@ do_trap_user(struct trapframe *frame)
 		userret(td, frame);
 		break;
 #if __has_feature(capabilities)
+	/* M-mode emulates alignment of non-CHERI instructions */
```

Comment:
Not sure if this needs to be guarded by __has_feature(capabilities).
I know M-mode should emulate all unaligned loads, but if we always add it unconditionally we can guard against broken firmware.

I guess the explicit cap/ddc loads/stores will raise sigbus now on HW without misaligned accesses?

Reply:
Yes, which is fine, only kernel assembly uses them.

Reply:
My thinking was that we should stay as close to upstream RISC-V. Delivering SIGBUS to a vanilla RISC-V violates the ABI, so if we're on such firmware the correct thing to do is panic because it's broken :)

Reply:
Yeah I was just wondering what happens if userspace were to use them. I guess technically M-mode should emulate them but SIGBUS is fine.